private/buf/bufwire/image_config_reader.go

```diff
@@ -146,15 +147,33 @@ func (i *imageConfigReader) getSourceOrModuleImageConfigs(
 	}
 	imageConfigs := make([]ImageConfig, 0, len(moduleConfigs))
 	var allFileAnnotations []bufanalysis.FileAnnotation
-	for _, moduleConfig := range moduleConfigs {
+	var directDeps []string
+	for idx, moduleConfig := range moduleConfigs {
+		for _, directDep := range moduleConfig.Config().Build.DependencyModuleReferences {
+			directDeps = append(directDeps, directDep.String())
+		}
+		directDepsMap := stringutil.SliceToMap(directDeps)
```

Comment:
Just a note: I'm not sure this should be something you should be doing at the `bufwire` level, and it feels wrong to make it an option for `moduleFileSetBuilder`

Reply:
Thanks! I wasn't very clear on what's the best place to do it, since the moduleFileSet builder only deals in terms of `GetDependenciesPins`, which reads them all from the `buf.lock` file, so basically all direct+transitive deps.

The only place where I can get the direct deps is wherever I get access to the build config, which I find it weird is not available from the module itself.